File: python/src/wdl_model/model/expressions/wdl_ternary_operation.py
from __future__ import annotations

from dataclasses import dataclass

from .wdl_expression import ComponentType, WdlExpression


@dataclass
class WdlTernaryOperation(WdlExpression):
    condition: WdlExpression | None = None
    trueValue: WdlExpression | None = None
    falseValue: WdlExpression | None = None

    def __str__(self) -> str:
        return f"if {self.condition or ''} then {self.trueValue or ''} else {self.falseValue or ''}"

    def componentType(self) -> ComponentType:
        return ComponentType.TERNARY_OPERATION

File: java/src/main/java/com/myriad/wdl/model/expressions/WdlTernaryOperation.java
package com.myriad.wdl.model.expressions;

import lombok.Getter;
import lombok.Setter;

public class WdlTernaryOperation implements WdlExpression {

  @Getter @Setter private WdlExpression condition;
  @Getter @Setter private WdlExpression trueValue;
  @Getter @Setter private WdlExpression falseValue;

  public WdlTernaryOperation() {}

  @Override
  public String toString() {
    return getClass().getSimpleName();
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.TERNARY_OP;
  }
}
